helm-gcs-publish/action.yml

```diff
@@ -1,15 +1,27 @@
 name: 'Helm GCS Publish'
 description: 'Published helm charts to GCS'
+inputs:
+  chart-path:
+    description: 'Path to helm chart directory'
+    required: true
+    default: './helm'
+  helm-gcs-repository: # Replaces $HELM_GCS_REPOSITORY env var. Assumes only one or other is set
+    description: 'Helm repository GCS URL'
+    required: false
+  helm-gcs-credentials: # Replaces $HELM_GCS_CREDENTIALS env var. Assumes only one or other is set
+    description: 'Credentials for writing to Helm repository'
+    required: false
 runs:
   using: "composite"
   steps: 
     - run: |
           CHART_VERSION=$(echo ${GITHUB_REF} | cut -d/ -f 3)
-          CHART_NAME=$(awk '/^name:/ {print $2}' ./helm/Chart.yaml)
+          CHART_NAME=$(awk '/^name:/ {print $2}' ${{ inputs.chart-path}}/Chart.yaml)
           export GOOGLE_APPLICATION_CREDENTIALS=${HOME}/helm-gcs-key.json
-          echo $HELM_GCS_CREDENTIALS > ${GOOGLE_APPLICATION_CREDENTIALS}
-          helm dependency update ./helm/
-          helm repo add helm-gcs $HELM_GCS_REPOSITORY
-          helm package --version ${CHART_VERSION} --app-version ${CHART_VERSION} ./helm/
+          echo ${HELM_GCS_CREDENTIALS}${{ inputs.helm-gcs-credentials }} > ${GOOGLE_APPLICATION_CREDENTIALS}
```

Comment:
Should we validate that only one of these is set? If both are set, will there be useful info to debug or the failure will be just wrong creds?

Reply:
Ideally, I just figured it wasn't worth the complexity because this is a temporary state so I can migrate the users of this in a backwards compatible way. The expected env variables will be removed after that's done.

Reply:
So working on some other actions came across the fact that `sh` supports default params... thought that was a `bash` feature. Anyway cleaned it up, inputs take precedence, falls back to env values now.